tools/prow-auto-bumper/main.go

```diff
@@ -0,0 +1,282 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// prow-auto-bumper finds stable Prow components version used by k8s,
+// and creates PRs updating them in knative/test-infra
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"math"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/google/go-github/github"
+	"github.com/knative/test-infra/shared/ghutil"
+)
+
+const (
+	org  = "kubernetes"
+	repo = "test-infra"
+	// PRHead is the head branch of k8s auto version bump PRs
+	// TODO(chaodaiG): using head branch querying is less ideal than using
+	// label `area/prow/bump`, which is not supported by Github API yet. Move
+	// to filter using this label once it's supported
+	PRHead = "k8s-ci-robot:autobump"
+	// PRBase is the base branch of k8s auto version bump PRs
+	PRBase = "master"
+	// Index for regex matching groups
+	imageImagePart = 1
+	imageTagPart   = 2
+	// Max difference away from target date
+	maxDelta = 2 * 24 // 2 days
+	// Safe duration is the smallest amount of hours a version stayed
+	safeDuration = 12 // 12 hours
+	maxRetry     = 3
+)
+
+var (
+	// matching            gcr.io /k8s-(prow|testimage)/(tide|kubekin-e2e|.*)    :vYYYYMMDD-HASH-EXTRA
+	imagePattern     = `\b(gcr\.io/k8s[a-z0-9-]{5,29}/[a-zA-Z0-9][a-zA-Z0-9_.-]+):(v[a-zA-Z0-9_.-]+)\b`
```

Comment:
Why the {5,29} limit? Wouldn't it be safer to use a non-greedy catch-all before /?